### ISSUE TITLE
false positive map_clone on .map(|x| *x) on Option<Deref<Target=T>>

### ISSUE THREAD

**Issue by drunkirishcoder** [rust-lang/rust-clippy#4824]
```
struct S(u16);

impl std::ops::Deref for S {
    type Target = u16;
    
    fn deref(&self) -> &u16 {
        &self.0
    }
}

fn main() {
    let s = Some(S(0));
    let _ = s.map(|v| *v);
}
```
```
    Checking playground v0.0.1 (/playground)
warning: You are using an explicit closure for copying elements
  --> src/main.rs:15:13
   |
15 |     let _ = s.map(|v| *v);
   |             ^^^^^^^^^^^^^ help: Consider calling the dedicated `copied` method: `s.copied()`
   |
   = note: `#[warn(clippy::map_clone)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#map_clone
```

suggested fix does not compile.

```
   Compiling playground v0.0.1 (/playground)
error[E0599]: no method named `copied` found for type `std::option::Option<S>` in the current scope
  --> src/main.rs:15:15
   |
15 |     let _ = s.copied();
   |               ^^^^^^ method not found in `std::option::Option<S>`
   |
   = note: the method `copied` exists but the following trait bounds were not satisfied:
           `&mut std::option::Option<S> : std::iter::Iterator`
```

**Comment by JohnTitor** [comment]
Fixed by #4937
